docs/RFCS/structured_configs.md

```diff
@@ -0,0 +1,137 @@
+- Feature Name: Configs for structured data
+- Status: draft
+- Start Date: 2015-08-19
+- RFC PR: [#2183](https://github.com/cockroachdb/cockroach/pull/2183)
+- Cockroach Issue: [#2090](https://github.com/cockroachdb/cockroach/issues/2090)
+
+# Summary
+
+This RFC describes the application of various configurations to the structured data
+representation. We address storage, distribution, and modification.
+
+# Motivation
+
+Structured data will soon be the only publicly accessible API. This requires a change
+in the way configurations are specified and applied, moving from prefix-based logic
+to databases and tables.
+
+The following configs currently exist:
+* Accounting: usage limits (bytes, counts). Not currently used.
+* Users: password storage for the web interface. Currently settable, but not used.
+* Zones: replication configuration. Used to make allocation decisions.
+
+# Prerequisites
+
+Complete implementation of this RFC depends on the following:
+* Gossip system database: [#2179](https://github.com/cockroachdb/cockroach/issues/2179)
+* Usage accounting: [#2185](https://github.com/cockroachdb/cockroach/issues/2185)
+
+# Detailed design
+
+## Storage
+
+We propose moving all configuration files to the `system` database as separate tables.
+
+The table schema depends on the configuration. Some (eg: `users`) apply globally,
+while some (eg: `accounting`, `zones`) apply to databases and tables.
+
+For each configuration, we propose the following schemas:
+
+#### Accounting
+
+The accounting table applies to databases and potentially to tables as well.
+Its contents have yet to be determined.
+
+The initial use of this table will be usage limits (a.k.a quotas). Actual usage
+storage is still TBD, see: [#2185](https://github.com/cockroachdb/cockroach/issues/2185).
+
+#### Users
+
+The users table is currently the only configuration applicable globally.
+It contains a mapping of username to password hash and salt.
+
+Schema:
+```SQL
+CREATE TABLE system.users (
+  "username"  CHAR primary key,
+  "hashed"    BLOB
+)
+```
+
+#### Zones
+
+The zone config applies to tables, with a database-level entry used as default.
+When a database is created, it will use a global default.
+When a table is created, it will inherit its database zone config.
+
+Schema:
+```SQL
+CREATE TABLE system.zones (
+  "target"           INT primary key,
+  "attributes"       CHAR,
+  "range_min_bytes"  INT,
+  "range_max_bytes"  INT,
+  "gc_ttl_seconds"   INT
+)
+```
+
+The `target` field is the ID of the database or table the config applies to.
```

Comment:
The ID space is indeed shared.
This is really just extra storage rather than being attached to the descriptors themselves. I think we'll want to have small wrappers (eg: `SHOW ZONES`) to automatically do joins between the various configs and the descriptor table to do id-to-name translation